packages/vscode-extension/src/generateSvg.ts

```diff
@@ -43,9 +58,42 @@ export async function generateSvg(args: {
     return;
   }
 
+  const workspace = vscode.workspace.workspaceFolders?.length ? vscode.workspace.workspaceFolders[0] : null;
   const parsedPath = __path.parse(editor.document.uri.fsPath);
-  const svgFileName = `${parsedPath.name}-svg.svg`;
-  const svgUri = editor.document.uri.with({ path: __path.join(parsedPath.dir, svgFileName) });
+
+  const tokens: Record<SettingsValueInterpolationToken, string | undefined> = {
+    "${workspaceFolder}": workspace?.uri.fsPath,
+    "${fileDirname}": parsedPath.dir,
+    "${fileExtname}": parsedPath.ext,
+    "${fileBasename}": parsedPath.base,
+    "${fileBasenameNoExtension}": parsedPath.name,
+  };
+
+  const fileExtensionNoDot = parsedPath.ext.replace(".", "");
+
+  const svgFilenameTemplateId = `kogito.${fileExtensionNoDot}.svgFilenameTemplate`;
+  const svgFilePathTemplateId = `kogito.${fileExtensionNoDot}.svgFilePath`;
+
+  const svgFilenameTemplate = vscode.workspace.getConfiguration().get(svgFilenameTemplateId, "");
+  const svgFilePathTemplate = vscode.workspace.getConfiguration().get(svgFilePathTemplateId, "");
+
+  if (__path.parse(svgFilenameTemplate).dir) {
+    vscode.window.showErrorMessage(
+      `The kogito.${fileExtensionNoDot}.svgFilenameTemplate setting should be a valid filename, without a path prefix. Current value: ${svgFilenameTemplate}`
+    );
+    return;
+  }
+
+  const svgFileName = interpolateSettingsValue({
+    tokens,
+    value: svgFilenameTemplate.length ? svgFilenameTemplate : "${fileBasenameNoExtension}-svg.svg",
```

Comment:
```suggestion
    value: svgFilenameTemplate? svgFilenameTemplate : "${fileBasenameNoExtension}-svg.svg",
```

Would that be better? I mean, `""` and `0` are falsy values, so it would be the same, but with less code? 😝

Reply:
Sure!